### PR TITLE
Cherry-pick Add and enforce contracts.allowSystemUseOfHapiSigs

### PR DESCRIPTION
**Pull request by lukelee-sl**
**Description**:
- Adds contracts.allowSystemUseOfHapiSigs property, a comma-separated list of zero or more HAPI operations represented by precompiles.
- Unless a precompile's operation appears in the list, it cannot activate signing requirements using the top-level `SignatureMap`.
- Validates end-to-end behavior in HapiSpecs.

### CODE REVIEW

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/SyntheticTxnFactory.java [hunk 1]
```diff
@@ -308,12 +308,15 @@ public TransactionBody.Builder createNonfungibleApproval(
...
+            var tokenAllowance = TokenAllowance.newBuilder()
                     .setTokenId(approveWrapper.tokenId())
                     .setOwner(Objects.requireNonNull(ownerId).toGrpcAccountId())
                     .setSpender(approveWrapper.spender())
-                    .setAmount(approveWrapper.amount().longValueExact())
-                    .build());
+                    .setAmount(approveWrapper.amount().longValueExact());
+            if (ownerId != null) {
+                tokenAllowance.setOwner(ownerId.toGrpcAccountId());
+            }
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/SyntheticTxnFactory.java:318]: We seem to be setting it twice on line 313 and 317

File: hedera-node/test-clients/src/main/resource/contract/contracts/DoTokenManagement/DoTokenManagement.sol [hunk 2]
```diff
@@ -0,0 +1,210 @@
+pragma solidity ^0.8.0;
+
+import "./hip-206/IHederaTokenService.sol";
```
Comment (Neeharika-Sompalli) [inline hedera-node/test-clients/src/main/resource/contract/contracts/DoTokenManagement/DoTokenManagement.sol:3]: seems `DoTokenManagement.json` is missing

Comment (lukelee-sl) [reply hedera-node/test-clients/src/main/resource/contract/contracts/DoTokenManagement/DoTokenManagement.sol:3]: added

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/SyntheticTxnFactory.java (hunk 1 above)
Comment (lukelee-sl) [reply hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/SyntheticTxnFactory.java]: fixed

**Review by Neeharika-Sompalli — DISMISSED** [review]
LGTM !

**Review by Nana-EC — APPROVED** [review]

**Pull request merged**